db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBProjectionManagementClient.java

```diff
@@ -11,44 +11,316 @@ private EventStoreDBProjectionManagementClient(EventStoreDBClientSettings settin
+     * Resetting a projection will truncate all emitted streams and re-emit all events.
+     * @param projectionName Name of the projection.
+     */
+    public CompletableFuture reset(final String projectionName) {
+        return this.reset(projectionName, ResetProjectionOptions.get());
+    }
+
+    /**
+     * Resets the projection, causing it to start again from the beginning of the stream/s it selects from.
+     * Resetting a projection will truncate all emitted streams and re-emit all events.
+     * @param projectionName Name of the projection.
+     * @param options Additional options.
+     */
+    public CompletableFuture reset(final String projectionName, ResetProjectionOptions options) {
+        return new ResetProjection(this.client, projectionName, options).execute();
+    }
+
+    /**
+     * Restarts the projection subsystem. This can be used to recover from certain kinds of errors.
+     */
+    public CompletableFuture restartSubsystem() {
+        return this.restartSubsystem(RestartProjectionSubsystemOptions.get());
+    }
+
+    /**
+     * Restarts the projection subsystem. This can be used to recover from certain kinds of errors.
+     * @param options Additional options.
+     */
+    public CompletableFuture restartSubsystem(RestartProjectionSubsystemOptions options) {
+        return new RestartProjectionSubsystem(this.client, options).execute();
+    }
+
+    /**
+     * Updates the projection's query and emit options.
+     * @param projectionName Name of the projection.
+     * @param query The JavaScript projection.
+     */
+    public CompletableFuture update(final String projectionName, final String query) {
+        return this.update(projectionName, query, null, UpdateProjectionOptions.get());
+    }
+
+    /**
+     * Updates the projection's query and emit options.
+     * @param projectionName Name of the projection.
+     * @param query The JavaScript projection.
+     * @param emitEnabled Whether the projection can emit events.
+     *                    If this is false then the projection will fault if linkTo or emit are used in the projection query.
+     */
+    public CompletableFuture update(final String projectionName, final String query, final Boolean emitEnabled) {
+        return this.update(projectionName, query, emitEnabled, UpdateProjectionOptions.get());
+    }
+
+    /**
+     * Updates the projection's query and emit options.
+     * @param projectionName Name of the projection.
+     * @param query The JavaScript projection.
+     * @param emitEnabled Whether the projection can emit events.
+     *                    If this is false then the projection will fault if linkTo or emit are used in the projection query.
+     * @param options Additional options.
+     */
+    public CompletableFuture update(final String projectionName, final String query, final Boolean emitEnabled, UpdateProjectionOptions options) {

```

Comment:
I would suggest pushing the `emitEnabled` parameter to `UpdateProjectionOptions` instead.